overlord/snapstate/handlers.go

```diff
@@ -1313,6 +1313,18 @@ func (m *SnapManager) doLinkSnap(t *state.Task, _ *tomb.Tomb) (err error) {
 	// Record the fact that the snap was refreshed successfully.
 	snapst.RefreshInhibitedTime = nil
 
+	tr := config.NewTransaction(st)
```

Comment:
can't we do this even earlier in phase2 itself? by definition if we are going ahead to refresh the snap it's not held anymore, no?

Reply:
Hmm, but we also need to do something on normal install / manual refresh, no? Otherwise if a snap has yet another update in the store very quickly after a manual update, we would consider it's previous hold start time?

Reply:
You are right but conceptually it feels confusing/late to do it here, we are already quite in the middle of refreshing the snap, differently from setting last refresh time, this is not about whether the refresh succeeded, by definition if the snap needed to be still held we would not even start refreshing or get this far.  Or am I confused?

I think it of two places that maybe make sense to do this instead:
* snapstate.doInstall around where we do softCheckNothingRunningForRefresh
* the prerequisites task handler perhaps  which is the first task nowadays of an install/refresh ?

Reply:
Done.